infra/getting_started/your_first_lambda_function.md

```diff
@@ -0,0 +1,187 @@
+# [Getting Started](./README.md) / Deploying your first AWS Lambda Function with Go and Terraform
+
+## Intro
+This tutorial focuses on deploying a "Hello, World" Go program as a Lambda function using Terraform. The first section will cover creating the program in Go and the second section will cover writing the Terraform script that will deploy the program into AWS. As a bonus, we'll cover how you can configure AWS CloudWatch to invoke your Lambda function on a recurring schedule.
+
+## Assumptions
+* You're at least on macOS Mojave 10.14
+* You have the following installed on your machine:
+  * Terraform
+  * AWS CLI
+  * Go
+* You have an AWS account.
+* You have awscli configured with you credentials.
+* You know some Terraform basics like "plan" and "apply".
+
+## Creating the Go program
+1. First things first, open up that command line.
+2. Create a directory wherever you desire and name it `hello-lambda`.
+3. Set your current directory to `hello-lambda`.
+4. AWS has kindly provided a Go library to develop Lambda functions. Let's install it.
+
+    ```
+    go get github.com/aws/aws-lambda-go/lambda
+    ```
+
+5. Create a new file called `main.go`. This will contain the code that runs in our Lambda function.
+6. Within `main.go`:
+
+    ```
+    package main
+
+    import (
+        "github.com/aws/aws-lambda-go/lambda"
+    )
+
+    func hello() (string, error) {
+        return "Hello, world!", nil
+    }
+
+    func main() {
+        lambda.Start(hello)
+    }
+
+    ```
+
+    Let's dissect this program for a second.
+
+    As you can see, we imported the AWS library we installed. We then utilize it to call `lambda.Start()`. This takes a handler function, effectively calling it, and passes Lambda requests through it. 
+
+    The handler in this case is our function `hello`. This handler is relatively simple. It doesn't take any internal Lambda requests and just returns the classic "Hello, world!".
+
+7. All that's left is to transform our Go program into something Lambda can actually run. We can accomplish this by creating a binary and storing it in a `.zip` file.
+
+    ```
+    GOOS=linux GOARCH=amd64 go build -o main main.go
+    zip main.zip main
+    ```
+
+## Deploying with Terraform
+1. In the `hello-lambda` directory, create a new file called `main.tf`.
+2. Open that new file up in your favorite text editor or IDE.
+3. Before we create our Lambda function, we need to tell Terraform the basics such as that we'll be deploying to AWS and what region we'll be deploying this function to.
+
+    ```
+    variable "aws_region" {
+        default = "us-west-2"
+    }
+
+    provider "aws" {
+        region = "${var.aws_region}"
+    }
+    ```
+
+4. Let's create that Lambda function. Add the following lines to the file.
+
+    ```
+    resource "aws_lambda_function" "test_lambda" {
```

Comment:
Does this (and the other resources) follow the naming described in the `infra/aws/naming.md` file? If not, why not? Not saying the naming doc is right (I pulled most of it out of the air) but we should aim to be consistent

Reply:
I'll make sure everything is consistent!